Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -422,15 +422,19 @@ static inline bool computeIsAlwaysOnLoggingAllowed(NetworkSession& session)
     if (parameters.networkActivityTracker)
         m_task.get()._nw_activity = parameters.networkActivityTracker->getPlatformObject();
 #endif
+
+    m_session->registerNetworkDataTask(*this);
 }
 
 NetworkDataTaskCocoa::~NetworkDataTaskCocoa()
 {
-    if (!m_task || !m_sessionWrapper)
-        return;
+    if (m_task && m_sessionWrapper) {
+        RELEASE_ASSERT(m_sessionWrapper->dataTaskMap.get([m_task taskIdentifier]) == this);
```

Comment:
Could pull `[m_task taskIdentifier]` out into its own local to avoid a potential double msg send.